junit-platform-launcher/src/main/java/org/junit/platform/launcher/listeners/discovery/AbortOnFailureLauncherDiscoveryListener.java

```diff
@@ -46,6 +65,23 @@ public void selectorProcessed(UniqueId engineId, DiscoverySelector selector, Sel
 		}
 	}
 
+	@Override
+	public void issueFound(UniqueId engineId, EngineDiscoveryIssue issue) {
+		issuesByEngineId.computeIfAbsent(engineId, __ -> new ArrayList<>()) //
+				.add(issue);
+	}
+
+	@Override
+	public void launcherDiscoveryFinished(LauncherDiscoveryRequest request) {
+		String configValue = request.getConfigurationParameters().get(
+			LauncherDiscoveryListeners.ABORT_ON_FAILURE_FAILURE_ISSUE_SEVERITIES_CONFIGURATION_PROPERTY_NAME).orElse(
+				LauncherDiscoveryListeners.DEFAULT_ABORT_ON_FAILURE_FAILURE_ISSUE_SEVERITIES);
+		Set<Severity> failureIssueSeverities = Arrays.stream(configValue.split(",")) //
+				.map(Severity::valueOf) // TODO case-insensitive parsing
+				.collect(toCollection(() -> EnumSet.noneOf(Severity.class)));
+		// TODO Collect errors and throw exception or log all issues
```

Comment:
:speech_balloon: The idea here is that the default `abortOnFailure` listener would treat the configured issue severities as failures and throw an exception containing all of them after the launcher has finished test discovery. In 5.13 the list would be empty (?) but in 6.0 we could make it strict and fail on deprecations and warnings.

Reply:
Alternatively, a discovery listener could collect all issues and report the engine descriptor as failed during execution.

Reply:
see https://github.com/junit-team/junit5/pull/4380#issuecomment-2715120110